mm2src/coins/eth/fee_estimation/eip1559/infura.rs

```diff
@@ -0,0 +1,127 @@
+use super::{EstimationSource, FeePerGasEstimated, FeePerGasLevel};
+use crate::eth::{Web3RpcError, Web3RpcResult};
+use crate::{wei_from_gwei_decimal, NumConversError};
+use mm2_err_handle::mm_error::MmError;
+use mm2_err_handle::prelude::*;
+use mm2_net::transport::slurp_url_with_headers;
+use mm2_number::BigDecimal;
+
+use http::StatusCode;
+use serde_json::{self as json};
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use url::Url;
+
+lazy_static! {
+    /// API key for testing
+    static ref INFURA_GAS_API_AUTH_TEST: String = std::env::var("INFURA_GAS_API_AUTH_TEST").unwrap_or_default();
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct InfuraFeePerGasLevel {
+    #[serde(rename = "suggestedMaxPriorityFeePerGas")]
+    pub suggested_max_priority_fee_per_gas: BigDecimal,
+    #[serde(rename = "suggestedMaxFeePerGas")]
+    pub suggested_max_fee_per_gas: BigDecimal,
+    #[serde(rename = "minWaitTimeEstimate")]
+    pub min_wait_time_estimate: u32,
+    #[serde(rename = "maxWaitTimeEstimate")]
+    pub max_wait_time_estimate: u32,
+}
+
+/// Infura gas api response
+/// see https://docs.infura.io/api/infura-expansion-apis/gas-api/api-reference/gasprices-type2
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+pub(crate) struct InfuraFeePerGas {
+    pub low: InfuraFeePerGasLevel,
+    pub medium: InfuraFeePerGasLevel,
+    pub high: InfuraFeePerGasLevel,
+    #[serde(rename = "estimatedBaseFee")]
+    pub estimated_base_fee: BigDecimal,
+    #[serde(rename = "networkCongestion")]
+    pub network_congestion: BigDecimal,
+    #[serde(rename = "latestPriorityFeeRange")]
+    pub latest_priority_fee_range: Vec<BigDecimal>,
+    #[serde(rename = "historicalPriorityFeeRange")]
+    pub historical_priority_fee_range: Vec<BigDecimal>,
+    #[serde(rename = "historicalBaseFeeRange")]
+    pub historical_base_fee_range: Vec<BigDecimal>,
+    #[serde(rename = "priorityFeeTrend")]
+    pub priority_fee_trend: String, // we are not using enum here bcz values not mentioned in docs could be received
+    #[serde(rename = "baseFeeTrend")]
+    pub base_fee_trend: String,
+}
```

Comment:
same as above

Reply:
backlogged in https://github.com/KomodoPlatform/komodo-defi-framework/issues/2207